eslint.config.js

```diff
@@ -0,0 +1,204 @@
+import tsParser from '@typescript-eslint/parser'
+import * as graphqlESLint from '@graphql-eslint/eslint-plugin'
+import eslintJsPlugin from '@eslint/js'
+
+const gqlSchemaPath = './src/typeDefs/schema.graphql'
+
+export default [
+    // GraphQL Linter for Operations and Fragments within code files
+    // {
+    //     files: [
+    //         '__tests__/*.ts',
+    //         'src/*.ts'
+    //     ],
+    //     // Setup processor for operations/fragments definitions on code-files
+    //     processor: graphqlESLint.processors.graphql,
+    //     languageOptions: {
+    //         parser: graphqlESLint,
+    //         parserOptions: {
+    //             globals: {
+    //                 schema: gqlSchemaPath
+    //             }
+    //         }
+    //     },
+    //     plugins: {
+    //         '@graphql-eslint': graphqlESLint
+    //     },
+    //     rules: {
+    //         '@graphql-eslint/no-anonymous-operations': 'error',
+    //         '@graphql-eslint/no-duplicate-fields': 'error',
+    //         '@graphql-eslint/no-unreachable-types': 'error'
+    //     },
+    //     ...graphqlESLint.flatConfigs['operations-recommended'].rules
+    // },
+    // GraphQL Linter for Schema files
+    {
+        // Setup GraphQL Parser
+        // files: ['src/**/*.{graphql,gql}'],
+        files: [
+            '__tests__/*.ts',
+            'src/*.ts'
+        ],
+        plugins: {
+            '@graphql-eslint': graphqlESLint
+        },
+        languageOptions: {
+            parser: graphqlESLint,
+            parserOptions: {
+                schema: gqlSchemaPath                
+            }
+        },
+        rules: {
+            // ...graphqlESLint.flatConfigs['schema-recommended'].rules,
+            // '@graphql-eslint/no-hashtag-description': 'off'
+            // '@graphql-eslint/no-anonymous-operations': 'error',
+            // '@graphql-eslint/no-duplicate-fields': 'error',
+            // '@graphql-eslint/no-unreachable-types': 'error'
+        }
+    },
+    // JSON Linter
+    // {
+    //     files: [
+    //         '*.json',
+    //         '*.json5'
+    //     ],
+    //     languageOptions: {
+    //         parser: jsonPlugin,
+    //         parserOptions: {
+    //             jsonSyntax: 'JSON'
+    //         }
+    //     },
+    //     plugins: {
+    //         json: jsonPlugin
+    //     },
+    //     rules: {
+    //         'json/no-dupe-keys': 'error',
+    //         'json/quotes': ['error', 'double', { avoidEscape: false }],
+    //         'json/quote-props': ['error', 'always', {}]
+    //         // ...jsonPlugin.configs['recommended-with-json'].rules,
+    //         // 'json/*': [
+    //         //     'error',
+    //         //     {
+    //         //         allowComments: true,
+    //         //         allowTrailingCommas: false,
+    //         //         allowTemplateLiterals: true,
+    //         //         duplicateKey: false
+    //         //     }
+    //         // ]
```

Comment:
Not quite sure what's going on with this eslint config and the code that's commented out twice. Maybe you meant to remove it?

Reply:
they are technically different. I want to get them working and not lose my progress, so I kept it in. (less important right now, but it's important to protect us for future schema changes and json errors.